### DISCUSSION TITLE
v5.56.0

### DISCUSSION THREAD

**Discussion by sokra** [webpack/webpack#14385] [General]
# Performance

* make DefinePlugin rebuild check more efficient performance and memory wise

<hr /><em>This discussion was created from the release <a href='https://github.com/webpack/webpack/releases/tag/v5.56.0'>v5.56.0</a>.</em>

**Comment by ghost** [comment]
Why do even the smallest things get a release. This makes no sense to me

**Reply by sokra** [reply]
Smaller releases make it easier for users to correlate problems to releases and that makes it easier for us to find problems.
Users also receive improvements faster.

Why are you worried by too small releases?

**Comment by danburzo** [comment]
The update unfortunately breaks my setup with the following error:

```
DefinePluginnode:internal/crypto/hash:105
    throw new ERR_INVALID_ARG_TYPE(
    ^

TypeError [ERR_INVALID_ARG_TYPE]: The "data" argument must be of type string or an instance of Buffer, TypedArray, or DataView. Received an instance of Set
    at new NodeError (node:internal/errors:370:5)
    at Hash.update (node:internal/crypto/hash:105:11)
    at BulkUpdateDecorator.update (/Users/danburzo/projects/<redacted>/node_modules/webpack/lib/util/createHash.js:51:14)
    at /Users/danburzo/projects/new-engine/<redacted>/webpack/lib/DefinePlugin.js:290:14
...
```

**Reply by alexander-akait** [reply]
Yes, bug, will be fixed in near future, thanks for report

**Reply by BoulosChaaya** [reply]
> The update unfortunately breaks my setup with the following error:
> 
> ```
> DefinePluginnode:internal/crypto/hash:105
>     throw new ERR_INVALID_ARG_TYPE(
>     ^
> 
> TypeError [ERR_INVALID_ARG_TYPE]: The "data" argument must be of type string or an instance of Buffer, TypedArray, or DataView. Received an instance of Set
>     at new NodeError (node:internal/errors:370:5)
>     at Hash.update (node:internal/crypto/hash:105:11)
>     at BulkUpdateDecorator.update (/Users/danburzo/projects/<redacted>/node_modules/webpack/lib/util/createHash.js:51:14)
>     at /Users/danburzo/projects/new-engine/<redacted>/webpack/lib/DefinePlugin.js:290:14
> ...
> ```

Goin back to Webpack version 5.55.1 seems to work pretty well

**Reply by sokra** [reply]
Should be fixed by https://github.com/webpack/webpack/releases/tag/v5.56.1

**Reply by danburzo** [reply]
I can confirm it's fixed in my case.
